### ISSUE TITLE
Discussion: SuiteCRM 7.8.x end-of-life

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#7294]
Dear SuiteCRM Community members:

As the scheduled **end-of-life (EOL) of the 7.8.x LTS version** is fast approaching, we’d like to open up a Community discussion on the challenges you currently have with upgrading away from 7.8.

This issue here on GitHub is just a way of calling attention to the discussion that is taking place on the Forums:

https://suitecrm.com/suitecrm/forum/suitecrm-7-0-discussion/24652-suitecrm-7-8-x-reaching-its-end-of-life

* **Please read the full thread on the forums and participate there!**

I will be doing a few mentions here (since the Forums unfortunately don't have a mention mechanism) to make sure people receive an email and take the opportunity to participate. Thanks in advance for any input.

**Comment by pgorod** [comment]
In no special order, and definitely not a complete list, these are some people who have made past contributions or participated in the Pull Request Party. We'd love to hear from you on the discussion linked above.

@horus68
@ApatheticCosmos
@mikebeck
@amariussi
@likhobory
@urdhvatech 
@Mausino 
@jakubpas
@JanSiero 
@lazka
@jsamelko 
@adriangibanelbtactic 
@ChangezKhan 
@Abuelodelanada 
@hieuhoanghd 
@LionelBino
@vladbar 
@sanchezfauste 
@adamjakab 
@isleshocky77 
@anhnhatuit 
@ghermans 
@eggsurplus 
@javitoron 
@sathishsengodan 
@jobst 
@bluquet 
@connorshea 
@gitbnw 
@604media

**Comment by Mausino** [comment]
Thanks for opportunity to add my opinion.. first i apologize for English.

2 main changes between version 7.8 and 7.10 are as you wrote 

- **Refined SuiteP theme - New Slimmed down SuiteP Design.**
- 4 SuiteP colour schemes - Pick a range of different colour schemes to suit your workspace.
- **New REST API - A new Rest API (v8) using the popular JSONAPI specifications.**
- Survey Management - A new module that will provide you the ability to create, design and send surveys to your customers.
- **Confirmed Opt-In - A new feature that has been introduced to assist CRM users with the European GDPR legislation.**
- **2 Factor Authentication**
- Improved Alerts Functionality
- Improved Password Management
- Additional password restrictions, logging to identify login attempts and enforce password requirements.
- Fail2Ban Logging Integration
- **New Suite of Unit Tests**
- Improved Email Performance
- **Email Fixes**

As you can see SaleAgility Team battle against in many levels...

I give reasons why uprade 
- GDPR legislation... sorry if you're from EU you have no choose if you want be in law 
- New Theme - as everybody know.. we are few years in world mobile first.
- REST API - if you want connect with world.. now other way.. now every new build applications are API first 
- Unit Tests - i hate them, you hate them.. but without the large projects will sink

I give you reasons why doesn't upgrade
- New Theme P
-- is unusable for my company. For 95% i'm using in mobile browser "web for pc view" and zooming... because if i see first column of default view + name of record...  i feel stupid...
-- please should have records in list view the + sign as is used for subpanels that i don't still need to go to detail view of record.. 
-- we need have own view for mobile and we must have option which columns we wan see on mobile (minimal for list view + sub-panels) and options what type of data will see when we click on + sign in front of record in row
-- when we want improve something in theme.. is nightmare.. no documentation.. (i am not talking about colors... but view ports or custom elements in theme)
- New Email module... 
-- introduced was in 7.9.. total disaster on begun. in 7.10. smaller disaster... when i put to GitHub word EMAIL i see that 209 issues have something with this module where are issues. 
-- in my heart i hoped that 7.11 will again big pressure to fix email child problems.... but tramtada... we introduced Google Calendars + Elastic Search ... and minimal pressure on email bugs.. many bug came back again, which was fixed in 7.10.
-- i told many times.. if you working with emails as one of most important module... **stay on 7.8 unsupported LTS as your business fired... because new email module 7.10 or 7.11 is only 40% of what you can do in 7.8.** do you want know what child problem are still in latest 7.10 or 7.11 version???
--- delete email... yes, you can't... why... don't know.. we have ACL in SuiteCRM for preventing on it if somebody scare.. sorry use other software or service
--- attachments in non imported emials - you want see what attachment you got... no you can't - fist import email and after you'll see that 90% of attachment are only images of signature of senders. you want see the attachments.. sorry use other software or service
--- filtering... you can't filter more inboxes or folders.. you can only default and no more.. sorry use other software or service
-- slow performance of query for emails... take tons of time to work with this module... if you want be more productive... sorry use other software or service
-- Run Email Notification is running but no notification is being sent... yes you can create meeting or task or what ever.. can you receive email notification? sorry..  
-- Auto Import of Emails not working... 

In last year started discussion about that many users are frustrated from system that community was "ignored" and SalesAgility created great event Pull Request Party (don't know when will next one.. hoped that will minimum ones in every quart of year) and changed some ways how communicate in this year with community . Now things moved forward and now we have milestones, better labels, and big pressure on API bug fixing. 

Maybe in few months will come SuiteCRM 8.. what we know? 
- Angular UI
- Service Enriched REST API
- General Framework Modernisation and Performance Improvements
- Fully automated testing for all new code
- Community Engagement

What we don't know
- Backwards Compatibility (where possible)
- New/Enhanced Extendable framework – easier third party development

If many developers will have answers on this 2 questions.. **they will know if need to spend time that will  spend on upgrade for 7.10 LTS or simply go direct to 8.1 LTS.**

**Comment by kelly-walker** [comment]
I concur with the notes below. The benefits to upgrading outweigh the negatives, especially with the noted security enhancements. It is not a small job to move from 7.8 with the theme.


Kind regards,

Kelly Walker
CEO




Process Flow Consulting Ltd.
toll-free: 877-832-3569
cell: 604-831-4129

kelly@processflow.ca
www.processflow.ca

Confidentiality Warning: This e-mail contains information intended only for the use of the individual or entity named above. If the reader of this e-mail is not the intended recipient or the employee or agent responsible for delivering it to the intended recipient, any dissemination, publication, or copying of this e-mail is strictly prohibited. The sender does not accept any responsibility for any loss, disruption or damage to your data or computer system that may occur while using data contained in, or transmitted with, this e-mail.

**Comment by hubyhuby** [comment]
Hi there,
I feel the same.
I have opened an issue on email test coverage.
In my opinion this is the biggest challenge we are facing : having a very stable email function.
Automated Test code coverage is an important to boot with : https://github.com/salesagility/SuiteCRM/issues/7299

**Comment by amariussi** [comment]
I agree with @Mausino on all points except the comments on GDPR: double opt-in is not compulsory and many companies have not adopted it so, in order to be compliant, SuiteCRM should only give an option to adopt it. The way in which it has been implemented makes it compulsory and is both against GDPR and the way in which several companies operate!

**Comment by connorshea** [comment]
The biggest problem thus far (upgrading from 7.9 to 7.10) was probably the theme changes, but for the most part they didn't break too much and only required a few changes for us. Thankfully, we implemented a system where if we modify any core files we mark the changes with a big comment block that explains the change and links to any relevant issues/PRs/commits.

```php
/**
 * CUSTOMIZATION NOTE
 * The change in this file was backported from 7.10.7, see <GitHub commit link> for more info.
 * We can use the upstream file once we've upgraded to 7.10.7+
 */
Code.goes_here()
/**
 * END CUSTOMIZATION NOTE
 */
```

This made the upgrade process – where I go through each changed file that we've customized and determine whether the upgraded file needs to be modified to preserve any of our customizations – a lot faster. Only took maybe 4-5 hours of work, compared to multiple days of tedious work when upgrading from 7.9.4 to 7.9.17 before this system was in place.

The test coverage is definitely a sore spot for me too, but ultimately the security fixes, new features, and upstream support are more important. We'd definitely appreciate improved test coverage so we can be more confident in the upgrades not having hidden bugs :)

I've also had some trouble determining whether issues were fixed in 7.10 or only in 7.11, e.g. #5173 which doesn't mention any specific version, and which I'm experiencing on 7.10.16.
